.ci/update-beats.yml

```
@@ -37,6 +37,21 @@ sources:
     spec:
       file: 'https://api.github.com/repos/elastic/beats/commits?sha={{ requiredEnv "BRANCH_NAME" }}&per_page=1'
       key: '.[0].sha'
+    transformers:
+      # substring 12 chars so it works for the condition
+      - findsubmatch:
+          pattern: ^(.{12}).*
+          captureindex: 1
+
+conditions:
+  is:
+    kind: file
```

NIT:

What do you think adding the name field to help with the messages in the PR description?

```suggestion
    name: Is version 'github.com/elastic/beats@{{ source "beats" }}' not updated in 'go.mod'?
    kind: file
```

For instance, this [run](https://github.com/elastic/golang-crossbuild/actions/runs/4354393920/jobs/7609633045#step:3:467) uses https://github.com/elastic/golang-crossbuild/blob/7971772e9dd1b54c660b1c32cabbb607c970621d/.ci/bump-golang.yml#L64

---

https://github.com/elastic/apm-server/pull/10436/commits/b36deef88372ba8a77dc85ea27312fdc7c05914f